lib/transformer.dart

```diff
@@ -84,14 +85,14 @@ class InitializeTransformer extends Transformer {
     if (primaryId == null) primaryId = transform.primaryInput.id;
     var newEntryPointId = new AssetId(primaryId.package,
         '${path.url.withoutExtension(primaryId.path)}.initialize.dart');
-    return transform.hasInput(newEntryPointId).then((exists) {
+    return transform.hasInput(newEntryPointId).then((exists) async{
```

Comment:
Yes it's better for sure.

Il mar 9 ago 2016, 16:41 Jacob MacDonald <notifications@github.com> ha
scritto:

> In lib/transformer.dart
> <https://github.com/dart-lang/initialize/pull/47#discussion_r74071637>:
>
> > @@ -84,14 +85,14 @@ class InitializeTransformer extends Transformer {
> >      if (primaryId == null) primaryId = transform.primaryInput.id;
> >      var newEntryPointId = new AssetId(primaryId.package,
> >          '${path.url.withoutExtension(primaryId.path)}.initialize.dart');
> > -    return transform.hasInput(newEntryPointId).then((exists) {
> > +    return transform.hasInput(newEntryPointId).then((exists) async{
>
> What about just changing this to use async/await to replace the then
> calls, instead of inside of them:
>
>     Future<AssetId> _buildBootstrapFile(Transform transform,
>        {AssetId primaryId}) {
>       if (primaryId == null) primaryId = transform.primaryInput.id;
>       var newEntryPointId = new AssetId(primaryId.package,
>           '${path.url.withoutExtension(primaryId.path)}.initialize.dart');
>       var exists = await transform.hasInput(newEntryPointId);
>       if (exists) {
>         transform.logger
>             .error('New entry point file $newEntryPointId already exists.');
>         return null;
>       }
>
>       var resolver = await _resolvers.get(transform, [primaryId]);
>       transform.addOutput(generateBootstrapFile(
>           resolver, transform, primaryId, newEntryPointId,
>           errorIfNotFound: _errorIfNotFound, plugins: plugins));
>       resolver.release();
>       return newEntryPointId;
>     }
>
> Ditto for this same issue in other cl's. It should work around the strong
> mode bugs and clean things up a bit.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dart-lang/initialize/pull/47/files/ecce1c1bee0470711f927b4831fb647e3c119621#r74071637>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTD-tYbD3nW-_KFwq7W02rWDT41vrIAks5qeJGagaJpZM4Jeg8P>
> .
>